### PR TITLE
Add Scala.js support

### LINKED ISSUE #25: Build for Scala JS

### PR DESCRIPTION
**Pull request by sbrunk**
This is an attempt to add `Scala.js` support to `scalacheck-toolbox` so that downstream projects can use it for tests running on Scala.js.

It adds Scala.js support to the `datetime` `combinators` modules. I left support for `magic` out for now as it would require changes in how `blns.txt` gets loaded.

Note that I've added [sbt-projectmatrix](https://github.com/sbt/sbt-projectmatrix) for sane cross-building but I had to remove `sbt-modules` along the way.

Fixes #25

### CODE REVIEW

**Comment by sbrunk** [comment]
It doesn't look like the check failure is related to my changes, but I'm not familiar enough with the project to know for sure.

File: build.sbt [hunk 1]
```diff
@@ -9,23 +15,53 @@ addCommandAlias("ci-publish", "github; ci-release")
...
-  .settings(libraryDependencies += "org.scalacheck" %% "scalacheck" % "1.14.3")
-
-lazy val `scalacheck-toolbox-combinators` = module
-  .settings(description := "Useful generic combinators for ScalaCheck")
-  .settings(libraryDependencies += "org.scalacheck" %% "scalacheck" % "1.14.3")
+  .settings(
+    libraryDependencies ++= Seq(
+      "org.scalacheck"         %%% "scalacheck"              % "1.14.3",
+      "org.scala-lang.modules" %%% "scala-collection-compat" % "2.2.0",
+      "joda-time"                % "joda-time"               % "2.10.6"
```
Comment (BenFradet) [inline build.sbt:43]: isn't this taken care of by the line below https://github.com/47degrees/scalacheck-toolbox/pull/166/files#diff-fdc3abdfd754eeb24090dbd90aeec2ceR48

**Review by BenFradet — COMMENTED** [review]
LGTM :+1:. Seems like the mdoc plugin now makes warnings fail the build. I'll try to fix it.

**Comment by BenFradet** [comment]
can you rebase against master, ci should be fixed

**Comment by BenFradet** [comment]
the warning which makes the build fails is related to scala 2.13.3

**Review by BenFradet — APPROVED** [review]
LGTM :+1: , there is still this q though https://github.com/47degrees/scalacheck-toolbox/pull/166/files#r492057195

File: build.sbt (hunk 1 above)
Comment (sbrunk) [reply build.sbt:44]: Yes. Good catch!

**Comment by sbrunk** [comment]
Running the tests now takes almost 2 hours compared to ~18 minutes before 🙀

The Scala.js tests run on node.js. Seems like they are by a magnitude slower especially on contrained CI runners.

File: build.sbt (hunk 1 above)
Comment (sbrunk) [reply build.sbt]: Done

**Comment by sbrunk** [comment]
Ok it seems like only `genPickFromMap*` in CombinatorProperties is crazy slow on JS. Would you mind if I reduce the number of required successful tests from `100000` to `10000` (only when running on Scala.js of course)?

https://github.com/47degrees/scalacheck-toolbox/blob/9dc3e812b22096b9187a785c2fcaaa10d51500a8/modules/scalacheck-toolbox-combinators/src/test/scala/com/fortysevendeg/scalacheck/combinators/CombinatorProperties.scala#L26-L27

**Comment by BenFradet** [comment]
fine with me as long as it's only for js :+1:

**Review by BenFradet — APPROVED** [review]
LGTM :+1:

**Review by fedefernandez — APPROVED** [review]
LGTM!

**Comment by sbrunk** [comment]
Dear maintainers,
do you have plans to cut a release in the near future? I'd love to use this library in a downstream project but I depend on a published version with Scala.js support. Thanks a lot!

**Comment by BenFradet** [comment]
yes, we should be releasing 0.4.0 today :+1:

**Pull request merged**